[Dart 3.10] What's new page update (#7005)
dart-lang/site-www: src/content/resources/whats-new.md

[dart-announce]: https://groups.google.com/a/dartlang.org/d/forum/announce
[Dart blog]: https://blog.dart.dev

## 3.10 release

_Released on: November 12, 2025_

This section lists notable changes made from August 14, 2025,
through November 12, 2025 to [dart.dev](https://dart.dev/docs).
For details about the 3.10 release of Dart,
check out the [3.10 announcement][] and the
[3.10 SDK changelog][3-10-changelog].

### Documentation updates {:.no_toc}

> *   Added a guide for the new [dot shorthands][] feature.

Reviewer: ```suggestion
*   Added a guide for the new [dot shorthands][] language feature.
```

*   Added a guide for the new [analyzer plugin][] feature.
*   Added a guide for the [`dart build`][] tool.

> *   Added a guide for the [`dart install`][] tool.

Reviewer: Maybe "reference" is more accurate for the type of content these pages have?

```suggestion
*   Added a reference for the [`dart build`][] tool.
*   Added a reference for the [`dart install`][] tool.
```

> *   Extended the [`@Deprecated` annotation][] content with new features.

Reviewer: Can we be a more specific here? It's not a new feature of the pre-existing `@Deprecated` annotation, but new annotations for deprecating other capabilities.

```suggestion
*   Documented the new [`@Deprecated` annotation][] varieties.
```

*   Expanded the [Glossary][] with new terms including "obviously typed",
    "closurization", "callback", "tree shaking", and "native asset".

>     (Thank you, [IldySilva](https://github.com/IldySilva))

Reviewer: Thanks again @IldySilva!! :D

### Site improvements {:.no_toc}

> *   Migrated core site infrastructure to the [Jaspr framework][Jaspr migration]

Reviewer: Consider cross linking Jaspr framework to https://jaspr.site. If you still want to link to the migration PR as well, maybe "Migrated core site infrastructure" can be.
